### PR TITLE
Nearest service

### PR DESCRIPTION
**Pull request by akhenakh**

### CODE REVIEW

**Comment by regeda** [comment]
Hey, update README too. `Nearest` is mentioned as unsupported.

File: near.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package osrm
+
+import geo "github.com/paulmach/go.geo"
+
+// NearRequest represents a request to the near method
+type NearRequest struct {
```
Comment (regeda) [inline near.go]: Rename `Near*` to `Nearest*` everywhere. We use a convention to name types according to OSRM API.

**Comment by akhenakh** [comment]
done

File: osrm.go [hunk 2]
```diff
@@ -100,3 +100,13 @@ func (o OSRM) Match(ctx context.Context, r MatchRequest) (*MatchResponse, error)
 	}
 	return &resp.MatchResponse, nil
 }
+
+// Nearest matches given GPS point to the nearest road network.
+// See https://github.com/Project-OSRM/osrm-backend/blob/master/docs/http.md#service-nearest for details.
```
Comment (regeda) [inline osrm.go]: the actual link on the doc is `https://github.com/Project-OSRM/osrm-backend/blob/master/docs/http.md#nearest-service`

File: nearest.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	Number   int
+}
+
+// NearestResponse represents a response from the nearest method
+type NearestResponse struct {
+	Waypoints []NearestWaypoint `json:"waypoints"`
+}
+
+// NearestWaypoint represents a nearest point on a nearest query
+type NearestWaypoint struct {
```
Comment (regeda) [inline nearest.go]: In the original doc, a waypoint response contains `nodes` (Array of OpenStreetMap node ids.):
```
{
   "waypoints" : [
      {
         "nodes": [
            2264199819,
            0
         ],
         "hint" : "KSoKADRYroqUBAEAEAAAABkAAAAGAAAAAAAAABhnCQCLtwAA_0vMAKlYIQM8TMwArVghAwEAAQH1a66g",
         "distance" : 4.152629,
         "name" : "Friedrichstraße",
         "location" : [
            13.388799,
            52.517033
         ]
      },
...
```
Could you add it as:
```
Nodes []uint64 `json:"nodes"`
```

**Comment by regeda** [comment]
Also, update the comment https://github.com/gojuno/go.osrm/blob/master/osrm.go#L18

**Comment by sbinq** [comment]
@akhenakh thanks for noticing docs issue! updated other links

File: nearest.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+type nearestResponseOrError struct {
+	responseStatus
+	NearestResponse
+}
+
+func (r NearestRequest) request() *request {
+	opts := options{}
+	opts.addInt("number", r.Number)
```
Comment (sbinq) [inline nearest.go]: Let's not make this param required - OSRM has default value here (`number integer >= 1 (default 1)`), and zero seems like not a valid input.

So could you please add zero value check like `if r.Number != 0 { opts.addInt(..) }`?

File: nearest.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
+package osrm
+
+import geo "github.com/paulmach/go.geo"
+
+// NearestRequest represents a request to the nearest method
+type NearestRequest struct {
+	Profile  string
+	GeoPath  GeoPath
+	Bearings []Bearing
+
```
Comment (sbinq) [inline nearest.go]: Extra minor: could you please remove extra space here? Just to be a bit more consistent with other requests around

File: nearest.go (hunk 4 above)
Comment (akhenakh) [reply nearest.go]: done `if r.Number >= 1`

File: nearest.go (hunk 5 above)
Comment (akhenakh) [reply nearest.go]: done

**Comment by akhenakh** [comment]
done

**Pull request merged**